### PR TITLE
Refactor local gpu scheduler

### PR DESCRIPTION
**Pull request by SparkSnail**
Fix gpu scheduler bug. Remove `node-nvidia-smi` package in localTrainingService, use `nni_gpu_tool.gpu_metrics_collector` to collector gpu information.

### CODE REVIEW

File: src/nni_manager/training_service/common/gpuData.ts [hunk 1]
```diff
@@ -58,3 +58,11 @@ export class GPUSummary {
         this.gpuInfos = gpuInfos;
     }
 }
+
+export const GPU_COLLECTOR_FORMAT: string =
```
Comment (yds05) [inline src/nni_manager/training_service/common/gpuData.ts]: How about GPU_INFO_COLLECTOR_FORMAT

Comment (yds05) [reply src/nni_manager/training_service/common/gpuData.ts]: You should also consider if the script works on Windows. 
However, it's not urgent for this hot fix. You can update the code for Windows after merging.

File: src/nni_manager/training_service/local/localTrainingServiceForGPU.ts [hunk 2]
```diff
@@ -69,9 +69,9 @@ class LocalTrainingServiceForGPU extends LocalTrainingService {
         }
     }
 
-    public cleanUp(): Promise<void> {
+    public async cleanUp(): Promise<void> {
         if (this.gpuScheduler !== undefined) {
-            this.gpuScheduler.stop();
+            await this.gpuScheduler.stop();
         }
```
Comment (yds05) [inline src/nni_manager/training_service/local/localTrainingServiceForGPU.ts]: Also need to remove metrics data file from tmp folder in cleanup like:

remove path.join(this.gpuMetricCollectorScriptFolder, 'gpu_metrics')

File: src/nni_manager/training_service/common/gpuData.ts (hunk 1 above)
Comment (SparkSnail) [reply src/nni_manager/training_service/common/gpuData.ts]: Changed to `GPU_INFO_COLLECTOR_FORMAT`. We may change the format to support windows in another PR.

File: src/nni_manager/training_service/local/localTrainingServiceForGPU.ts (hunk 2 above)
Comment (SparkSnail) [reply src/nni_manager/training_service/local/localTrainingServiceForGPU.ts]: Add remove temp metrics files function.

**Comment by chicm-ms** [comment]
Is the purpose to fix the problem Mao mentioned?   If it is only refatoring, please do it on master.

**Comment by SparkSnail** [comment]
> Is the purpose to fix the problem Mao mentioned? If it is only refatoring, please do it on master.

This pr is aimed to fix gpu scheduler bug which Mao mentioned, it is not only refactoring.

File: src/nni_manager/training_service/local/gpuScheduler.ts [hunk 3]
```diff
@@ -321,51 +83,18 @@ class GPUScheduler {
...
-        if(gpuNumber == 1) {
-            const embededGPUSummary = <nodeNvidiaSmi.EmbededGPUSummary>data.nvidia_smi_log.gpu;
-            gpuInfos.push(this.convertGPUSummaryToInfo(embededGPUSummary));
-        } else {
-            const embededGPUSummaryArray = <nodeNvidiaSmi.EmbededGPUSummary[]>data.nvidia_smi_log.gpu;
-            gpuInfos = embededGPUSummaryArray.map(embededGPUSummary => this.convertGPUSummaryToInfo(embededGPUSummary));
+    private async updateGPUSummary() {
+        const cmdresult = await cpp.exec(`tail -n 1 ${path.join(this.gpuMetricCollectorScriptFolder, 'gpu_metrics')}`);
+        if(cmdresult && cmdresult.stdout) {
+            this.gpuSummary = <GPUSummary>JSON.parse(cmdresult.stdout);
```
Comment (QuanluZhang) [inline src/nni_manager/training_service/local/gpuScheduler.ts]: is there error log, if gpu metric is not correctly read?

Comment (SparkSnail) [reply src/nni_manager/training_service/local/gpuScheduler.ts]: added log information.

**Pull request merged**